zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessAutoConfiguration.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.zakenapi.uploadprocess
+
+import com.ritense.processdocument.service.ProcessDocumentAssociationService
+import com.ritense.processdocument.service.ProcessDocumentService
+import com.ritense.resource.service.TemporaryResourceStorageService
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.annotation.Order
+
+@Configuration
+class UploadProcessAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean(ResourceUploadedEventListener::class)
+    fun resourceUploadedEventListener(
+        resourceService: TemporaryResourceStorageService,
+        processDocumentService: ProcessDocumentService,
+    ): ResourceUploadedEventListener {
+        return ResourceUploadedEventListener(
+            resourceService,
+            processDocumentService,
+        )
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(UploadProcessResource::class)
+    fun uploadProcessResource(
+        processDocumentAssociationService: ProcessDocumentAssociationService
+    ): UploadProcessResource {
+        return UploadProcessResource(processDocumentAssociationService)
+    }
+
+    @Order(360)
```

Comment:
Why is the order needed?

Reply:
Im not sure. We have the @Order above every HttpSecurityConfigurer and the order does matter. See:
https://github.com/valtimo-platform/valtimo-backend-libraries/blob/development/core/src/main/java/com/ritense/valtimo/autoconfigure/HttpSecurityAutoConfiguration.java#L109

But I dont know how to decide which order is the correct one